python/lsst/afw/image/exposure.cc

```diff
@@ -24,13 +24,83 @@
+//            py::arg("fitsFile"), py::arg("bbox")=lsst::afw::geom::Box2I(), py::arg("origin")=PARENT, py::arg("conformMasks")=false);
+    cls.def(py::init<Exposure<PixelT> const &, bool>(),
+            py::arg("other"), py::arg("deep")=false);
+//    cls.def(py::init<Exposure<PixelT> const &, lsst::afw::geom::Box2I const&, ImageOrigin, bool>(),
+//            py::arg("other"), py::arg("bbox"), py::arg("origin")=PARENT, py::arg("deep")=false);
+
+
+    /* Members */
+    cls.def("getMaskedImage", (typename Exposure<PixelT>::MaskedImageT (Exposure<PixelT>::*)()) &Exposure<PixelT>::getMaskedImage);
+    cls.def("getWcs", (PTR(Wcs) (Exposure<PixelT>::*)()) &Exposure<PixelT>::getWcs);
+//    cls.def("getDetector", &Exposure<PixelT>::getDetector);
+    cls.def("getFilter", &Exposure<PixelT>::getFilter);
+    cls.def("getMetadata", &Exposure<PixelT>::getMetadata);
+    cls.def("setMetadata", &Exposure<PixelT>::setMetadata);
+    cls.def("getWidth", &Exposure<PixelT>::getWidth);
+    cls.def("getHeight", &Exposure<PixelT>::getHeight);
+    cls.def("getDimensions", &Exposure<PixelT>::getDimensions);
+    cls.def("getX0", &Exposure<PixelT>::getX0);
+    cls.def("getY0", &Exposure<PixelT>::getY0);
+    cls.def("getXY0", &Exposure<PixelT>::getXY0);
+    cls.def("getBBox", &Exposure<PixelT>::getBBox);
+    cls.def("setXY0", &Exposure<PixelT>::setXY0);
+    cls.def("setMaskedImage", &Exposure<PixelT>::setMaskedImage);
+    cls.def("setWcs", &Exposure<PixelT>::setWcs);
+//    cls.def("setDetector", &Exposure<PixelT>::setDetector);
+    cls.def("setFilter", &Exposure<PixelT>::setFilter);
+//    cls.def("setCalib", &Exposure<PixelT>::setCalib);
+//    cls.def("getCalib", &Exposure<PixelT>::getCalib);
+    cls.def("setPsf", &Exposure<PixelT>::setPsf);
+    cls.def("getPsf", (PTR(lsst::afw::detection::Psf) (Exposure<PixelT>::*)()) &Exposure<PixelT>::getPsf);
+    cls.def("hasPsf", &Exposure<PixelT>::hasPsf);
+    cls.def("hasWcs", &Exposure<PixelT>::hasWcs);
+    cls.def("getInfo", (PTR(ExposureInfo) (Exposure<PixelT>::*)())&Exposure<PixelT>::getInfo);
+
+    cls.def("writeFits", (void (Exposure<PixelT>::*)(std::string const &) const) &Exposure<PixelT>::writeFits);
+    cls.def("writeFits", (void (Exposure<PixelT>::*)(lsst::afw::fits::MemFileManager &) const) &Exposure<PixelT>::writeFits);
+    cls.def("writeFits", (void (Exposure<PixelT>::*)(lsst::afw::fits::Fits &) const) &Exposure<PixelT>::writeFits);
+
+    cls.def_static("readFits", (Exposure<PixelT> (*)(std::string const &)) Exposure<PixelT>::readFits);
+    cls.def_static("readFits", (Exposure<PixelT> (*)(lsst::afw::fits::MemFileManager &)) Exposure<PixelT>::readFits);
+}
+
 PYBIND11_PLUGIN(_exposure) {
     py::module mod("_exposure", "Python wrapper for afw _exposure library");
 
+    declareExposure<int>(mod, "I");
+    declareExposure<float>(mod, "F");
+    declareExposure<double>(mod, "D");
+    declareExposure<std::uint16_t>(mod, "U");
```

Comment:
You never declare ExposureL, but it is defined in SWIG. Make sure that pybind11 isn't casting longs to ints.